src/libraries/System.Net.Http.Json/src/System/Net/Http/Json/HttpContentJsonExtensions.cs

```diff
@@ -0,0 +1,84 @@
+﻿// Licensed to the .NET Foundation under one or more agreements.
+// The .NET Foundation licenses this file to you under the MIT license.
+// See the LICENSE file in the project root for more information.
+
+using System.Diagnostics;
+using System.IO;
+using System.Net.Mime;
+using System.Text;
+using System.Text.Json;
+using System.Threading;
+using System.Threading.Tasks;
+
+namespace System.Net.Http.Json
+{
+    public static class HttpContentJsonExtensions
+    {
+        public static Task<object?> ReadFromJsonAsync(this HttpContent content, Type type, JsonSerializerOptions? options = null, CancellationToken cancellationToken = default)
+        {
+            return ReadFromJsonAsyncCore(content, type, options, cancellationToken);
+        }
+
+        public static Task<T> ReadFromJsonAsync<T>(this HttpContent content, JsonSerializerOptions? options = null, CancellationToken cancellationToken = default)
+        {
+            return ReadFromJsonAsyncCore<T>(content, options, cancellationToken);
+        }
+
+        private static async Task<object?> ReadFromJsonAsyncCore(HttpContent content, Type type, JsonSerializerOptions? options, CancellationToken cancellationToken)
+        {
+            Stream contentStream = await GetJsonStreamFromContentAsync(content, cancellationToken).ConfigureAwait(false);
+            return await JsonSerializer.DeserializeAsync(contentStream, type, options, cancellationToken);
+        }
+
+        private static async Task<T> ReadFromJsonAsyncCore<T>(HttpContent content, JsonSerializerOptions? options, CancellationToken cancellationToken)
+        {
+            Stream contentStream = await GetJsonStreamFromContentAsync(content, cancellationToken).ConfigureAwait(false);
+            return await JsonSerializer.DeserializeAsync<T>(contentStream, options, cancellationToken);
```

Comment:
In the absence of user specified options, these should likely default to

```
JsonSerializerOptions
{ 
   PropertyNamingPolicy = JsonNamingPolicy.CamelCase, 
   PropertyNameCaseInsensitive = true,
}
```

This would make it the most compatible for content produced by an [MVC application](https://github.com/dotnet/aspnetcore/blob/master/src/Mvc/Mvc.Core/src/JsonOptions.cs#L15-L29). Users migrating from the WebAPI \ Newtonsoft.Json based extensions would also expect this.

Reply:
It's not just about consistency with ASP.NET tech. Using camel casing for JSON is the norm for web. We went through this casing debate for the various Web API stacks we've shipped over the years and received loads of feedback that folks expect the default casing to be camel cased when doing web development.

Reply:
It is actually in the spec to expose a new method that returns the options used by ASP.NET. https://github.com/dotnet/designs/blob/master/accepted/2020/json-http-extensions/json-http-extentions.md#default-serialization-options

@terrajobst should I include those API changes in this PR?